### DISCUSSION TITLE
Notes Turning Blue in Playback Code?

### DISCUSSION THREAD

**Discussion by DLu** [musescore/MuseScore#12151] [Q&A] [answered]
I'm looking to play with a new feature in playback, but my initial search for the relevant code has come up empty. 

Where can I find the code that handles the visual elements of playback, i.e. turning the notes blue, displaying the vertical bar, etc? I don't even know the names for these features. 

I tracked it as far as `IPlaybackController` before I got lost.

**Comment by cbjeukendrup** [comment, marked as answer]
The code for drawing the blue bar is in the PlaybackCursor class. The feature of notes being colored blue is not (yet) present in MuseScore 4.

**Reply by DLu** [reply]
:bow:
